streams/src/main/java/org/apache/kafka/streams/processor/internals/RestoringTasks.java

```diff
@@ -18,6 +18,6 @@
 
 import org.apache.kafka.common.TopicPartition;
 
-public interface RestoringTasks {
-    Task restoringTaskFor(final TopicPartition partition);
+public interface RestoringTasks<T extends Task> {
```

Comment:
Only `StreamsTask` will ever be restored. Perhaps we can remove the generics here?